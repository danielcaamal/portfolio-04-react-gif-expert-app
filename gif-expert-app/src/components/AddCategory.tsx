import { useState } from 'react';


export const AddCategory = ({
    onNewCategory = (newCategory: string) => {},
    // setCategories 
}) => {

    const [inputValue, setInputValue] = useState<string>("");

    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        setInputValue(e.target.value);
    }

    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // setCategories((cat:string[]) => ([...cat, inputValue]));
        onNewCategory(inputValue);
        setInputValue("");
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Search a Gif"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}