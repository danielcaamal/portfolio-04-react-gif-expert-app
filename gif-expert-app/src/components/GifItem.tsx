export const GifItem = (
    { id, title, url }: { id: string; title: string; url: string}
    ) => {

    const getTitle = title.trim() != "" ? title : "N/A";

    return (
        <div className="card">
            <img src={url} alt={title} id="gif-item-img"/>
            <p id="gif-item-title">{getTitle}</p>
        </div>
    );
}