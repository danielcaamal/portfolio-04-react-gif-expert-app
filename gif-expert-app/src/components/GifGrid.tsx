
import { GiphInterface } from "../interfaces";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category } : { category: string }) => {

    const {images, isLoading} = useFetchGifs(category);
    
    return (
        <>
            <h3 id="gif-grid-title">{category}</h3>
            {isLoading && (<h2 id="gif-grid-loading">Loading...</h2>)}
            <div className="card-grid" id="gif-grid-card">
                {images.map((image: GiphInterface) => (
                    <GifItem 
                        key={image.id}
                        {...image}
                    />
                ))}
            </div>
        </>
    )       
}