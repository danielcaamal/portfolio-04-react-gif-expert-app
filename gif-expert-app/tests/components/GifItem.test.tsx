import React from 'react';
import { render } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';

describe('GifItem', () => {
    const url = 'https://test.com/';
    const title = 'test';
    const id = '123';

    test('should match with snapshot', () => {
        const { container } = render(<GifItem
            id={id}
            title={title}
            url={url}
        />);
        expect(container).toMatchSnapshot();
    });

    test('should show the image with an url and alt send', () => {
        const { container } = render(<GifItem
            id={id}
            title={title}
            url={url}
        />);
        const { src, alt } = container.querySelector('#gif-item-img') as HTMLImageElement;
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show the title send', () => {
        const { container } = render(<GifItem
            id={id}
            title={title}
            url={url}
        />);
        expect(container.querySelector('#gif-item-title')?.textContent).toBe(title);
    });
});