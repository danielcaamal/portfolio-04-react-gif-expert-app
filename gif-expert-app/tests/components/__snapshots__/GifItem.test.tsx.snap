// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      id="gif-item-img"
      src="https://test.com/"
    />
    <p
      id="gif-item-title"
    >
      test
    </p>
  </div>
</div>
`;
